tests/Guzzle5ExceptionTest.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace Http\Adapter\Tests;
+
+use GuzzleHttp\Exception as GuzzleExceptions;
+use Http\Adapter\Guzzle5HttpAdapter;
+
+/**
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+class Guzzle5ExceptionTest extends \PHPUnit_Framework_TestCase
+{
+    public function testGetException()
+    {
+        $request = $this->getMock('Psr\Http\Message\RequestInterface');
+        $guzzleRequest = $this->getMock('GuzzleHttp\Message\RequestInterface');
+        $body = $this->getMock('GuzzleHttp\Stream\StreamInterface');
+        $body->expects($this->any())->method('detach')->willReturn('message body');
+        $guzzleResponse = $this->getMock('GuzzleHttp\Message\ResponseInterface');
+        $guzzleResponse->expects($this->any())->method('getStatusCode')->willReturn('400');
+        $guzzleResponse->expects($this->any())->method('getHeaders')->willReturn(array());
+        $guzzleResponse->expects($this->any())->method('getProtocolVersion')->willReturn('1.1');
+        $guzzleResponse->expects($this->any())->method('getBody')->willReturn($body);
+
+
+        $adapter = new Guzzle5HttpAdapter();
+        $method  = new \ReflectionMethod('Http\Adapter\Guzzle5HttpAdapter', 'handleException');
+        $method->setAccessible(true);
+
+        $outputException = $method->invoke($adapter, new GuzzleExceptions\ConnectException('foo', $guzzleRequest), $request);
```

Comment:
If something, then I would mock the client an throw actual exceptions so we can test the public API instead of using reflection.